### PR TITLE
implemented Authorization

### PR DESCRIPTION
**Pull request by matviichuk92**

### CODE REVIEW

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/all", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/all", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/order/complete", Set.of(Role.RoleName.USER));
```
Comment (AnaLanda) [inline src/main/java/online/shop/web/filter/AuthorizationFilter.java]: What about these URLs:
`/users/delete`
`/shopping-cart/delete`
`/orders/delete`
`/products/delete`
?

File: src/main/webapp/WEB-INF/views/access-denied.jsp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Access denied page</title>
+</head>
+<body>
+<h1>Access denied, sweetheart!</h1>
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/access-denied.jsp:7]: 😆

**Review by AnaLanda — COMMENTED** [review]
Looks great! I think you could add more URLs to the Set of protected URLs, though.

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java (hunk 1 above)
Comment (matviichuk92) [reply src/main/java/online/shop/web/filter/AuthorizationFilter.java]: AnaLanda , thanks a lot!

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/all", Set.of(Role.RoleName.USER));
```
Comment (dmytro-hryhoruk) [inline src/main/java/online/shop/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("/products/all", Set.of(Role.RoleName.USER));
```

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", Set.of(Role.RoleName.ADMIN));
```
Comment (dmytro-hryhoruk) [inline src/main/java/online/shop/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("/products/add", Set.of(Role.RoleName.ADMIN));
```

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        protectedUrls.put("/products/delete", Set.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/all", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart", Set.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/product/add", Set.of(Role.RoleName.USER));
```
Comment (dmytro-hryhoruk) [inline src/main/java/online/shop/web/filter/AuthorizationFilter.java]:
```suggestion
        protectedUrls.put("/shopping-carts/products/add", Set.of(Role.RoleName.USER));
```

File: src/main/java/online/shop/web/filter/AuthorizationFilter.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import online.shop.service.UserService;
+
+public class AuthorizationFilter implements Filter {
+    private static final String USER_ID = "userId";
+    private static final Injector injector = Injector.getInstance("online.shop");
+    private final UserService userService = (UserService) injector.getInstance(UserService.class);
+    private Map<String, Set<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/online/shop/web/filter/AuthorizationFilter.java:27]: user orders?

Comment (matviichuk92) [reply src/main/java/online/shop/web/filter/AuthorizationFilter.java:27]: There is _/orders/all_(see all user's orders) exist in the previous commit. Also, _/orders_ for admin(see all orders and ability to delete).

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**